### PR TITLE
Add changelog for v1.10.0

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-1.10.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since 1.9.0
+
+## Changes by Kind
+
+### Feature
+
+- Add Modify Volume Support for VolumeAttributesClass, this requires:
+  1. The feature gate VolumeAttributesClass and API to be enabled in Kubernetes cluster
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-1.10.md:12]: Also note this requires a 1.29 cluster

Comment (sunnylovestiramisu) [reply CHANGELOG/CHANGELOG-1.10.md:12]: It is listed in ### Uncategorized

Do we want to categorize it as feature from now on?

File: CHANGELOG/CHANGELOG-1.10.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  2. The feature gate VolumeAttributesClass feature gate enabled in external-resizer ([#351](https://github.com/kubernetes-csi/external-resizer/pull/351), [@sunnylovestiramisu](https://github.com/sunnylovestiramisu))
```
Comment (xing-yang) [inline CHANGELOG/CHANGELOG-1.10.md:13]: While looking at this PR, I noticed that the README needs to be updated:
https://github.com/kubernetes-csi/external-resizer/pull/351/files#r1468258087

File: CHANGELOG/CHANGELOG-1.10.md (hunk 1 above)
Comment (sunnylovestiramisu) [reply CHANGELOG/CHANGELOG-1.10.md]: Updated.

**Pull request merged**